sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -114,7 +114,16 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
     }
 
     (file: PartitionedFile) => {
-      val parser = new JacksonParser(actualSchema, parsedOptions)
+      // SPARK-21610: when the `requiredSchema` only contains `_corrupt_record`,
```

Comment:
Hm, I understood it fixes the issue described in the JIRA but won't this introduce casting tries for all columns when the requested schema is empty? I think this one is a rather band-aid fix. I mean, I think if the `actualSchema` has few columns selected from many columns, it'd introduce similar problem again ..